pkg/minishift/registration/redhat_test.go

```diff
@@ -75,22 +94,14 @@ func TestRedHatRegistratorRegister(t *testing.T) {
 	}
 	commander := provision.GenericSSHCommander{Driver: d}
 	registrator := NewRedHatRegistrator(commander)
-	s.CommandToOutput["sudo subscription-manager version"] = `server type: RedHat Subscription Management`
-	if err := registrator.Register(param); err != nil {
-		t.Fatal("Distribution shouldn't have to register")
-	} else {
-		cmd := "sudo subscription-manager register --auto-attach --username foo --password foo"
-		if _, ok := s.Commands[cmd]; ok{
-			t.Fatalf("Expected command: %s", cmd)
-		}
-	}
+
 	s.CommandToOutput["sudo subscription-manager version"] = `server type: This system is currently not registered.`
 	if err := registrator.Register(param); err != nil {
 		t.Fatal("Distribution should able to register")
 	} else {
 		cmd := "sudo subscription-manager register --auto-attach --username foo --password foo "
```

Comment:
Ok, let's try this a different way. Say your production code changes and the registration command indeed needs changing. I think we can agree that in this case the test will fail. What will you have to update to make the test pass again? The value of _cmd_, right? You can change _expectedCMDRegistration_ as much as you like, but to get the test pass again, you need to change _cmd_ to match the **expectation** of _Register_ in _redhat.go_.